src/wasm/wasm-emscripten.cpp

```diff
@@ -1133,17 +1225,15 @@ std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
   std::stringstream meta;
   meta << "{\n";
 
-  AsmConstWalker emAsmWalker = fixEmAsmConstsAndReturnWalker(wasm);
```

Comment:
Interesting. This now assumes `fixEmAsm` was ran beforehand. In the event that it wasn't, it simply omits the "asmConsts" metadata because `asmSigsForCode` will be empty.

This is good, because it makes generating metadata not side-effecting. Nice.